### PR TITLE
No validation fix

### PR DESCRIPTION
**Pull request by tkornuta-nvidia**
- [x] "Fixes" the problem appearing when one will try to train a model without providing the validation set
(generally the solution "disables" the `is_overriden` mechanism that PTL trainer uses)
 - [x] Example from CV collection is once again operational
 - [x] added L0 image classification test to jenkins - integration script using PTL trainer fast_dev_run

### CODE REVIEW

File: nemo/core/classes/modelPT.py [hunk 1]
```diff
@@ -312,9 +312,16 @@ def val_dataloader(self):
         if self._validation_dl is not None:
             return self._validation_dl
 
-    # def test_dataloader(self):
-    #     if self._test_dl is not None:
-    #         return self._test_dl
+    def validation_step(self, batch, what_is_this_input):
```
Comment (tkornuta-nvidia) [inline nemo/core/classes/modelPT.py:315]: should we also add such a "patch" to training_step?

use-case: when one will try to write his own pytorch training loop?

Comment (titu1994) [reply nemo/core/classes/modelPT.py:315]: I think it's not necessary. At the bare minimum all of our Nemo models will have train step defined, and users can define their own train loop inside ptl too. But it's fine to have either way. I'll leave it to the reviewers.

Comment (tkornuta-nvidia) [reply nemo/core/classes/modelPT.py:315]: wait... train loop inside PTL? what do you mean? would like to learn more, let's take that offline.

File: Jenkinsfile [hunk 2]
```diff
@@ -311,6 +311,25 @@ pipeline {
...
+        }
+      }
+      failFast true
+      parallel {
+        stage ('MNIST image classification with LeNet-5 Test - on CPU') {
+          steps {
+            sh 'cd examples/cv && python mnist_lenet5_image_classification_pure_lightning.py trainer.gpus=0 \
+            trainer.fast_dev_run=true model.dataset.data_folder=/home/TestData'
+            sh 'rm -rf examples/cv/outputs'
+          }
```
Comment (titu1994) [inline Jenkinsfile:328]: This should change exp_manager.root_dir to some custom path - say examples/CV/lenet_mnist and the rm below should remove that specific file. This is for future proofing against parallel ci runs which all target and read/write to same default root for.

File: nemo/collections/cv/models/mnist_lenet5.py [hunk 3]
```diff
@@ -100,9 +96,6 @@ def __init__(self, cfg: MNISTLeNet5Config = MNISTLeNet5Config()):
         # This will be set by setup_test_data
         self._test_dl = None
```
Comment (titu1994) [inline nemo/collections/cv/models/mnist_lenet5.py:98]: There's no need for these three _dl lines anymore.

File: Jenkinsfile (hunk 2 above)
Comment (tkornuta-nvidia) [reply Jenkinsfile:328]: a) I have copied this from the other jenkins examples. b) This "test" runs on CPU.

Comment (titu1994) [reply Jenkinsfile:328]: 1) Look at ASR Jenkins block 2) Why is it even on cpu anyway ?

Comment (tkornuta-nvidia) [reply Jenkinsfile:327]: still, removed 'rm'

File: nemo/collections/cv/models/mnist_lenet5.py (hunk 3 above)
Comment (tkornuta-nvidia) [reply nemo/collections/cv/models/mnist_lenet5.py]: removed

File: Jenkinsfile (hunk 2 above)
Comment (okuchaiev) [reply Jenkinsfile:327]: All CI machines have GPUs. I'm curious too - why not run this on GPU? Or GPU and CPU if you wish to test both. But only CPU seems strange.

Comment (titu1994) [reply Jenkinsfile:327]: Removing RM will cause weights to persist in the CI env, and eventually it will run out of space. As I said, you should set the root_dir environment of exp_manager to a custom path, and rm -rf just that dir.

File: nemo/core/classes/modelPT.py (hunk 1 above)
Comment (titu1994) [reply nemo/core/classes/modelPT.py:315]: @okuchaiev Any preference? I don't mind having a default traon_step() or not, so I'll defer to your opinion here.

Comment (okuchaiev) [reply nemo/core/classes/modelPT.py:315]: I would add this patch to training_step as well to make it easier to use without lightning's trainer, but I'd leave it to @tkornuta-nvidia  to make a call here

Comment (tkornuta-nvidia) [reply nemo/core/classes/modelPT.py]: ok, added

File: Jenkinsfile (hunk 2 above)
Comment (tkornuta-nvidia) [reply Jenkinsfile:327]: This is simple: I am saving like 1s by not moving model to GPU takes time.

Why should I bother with GPU when I have a simple model and I just want to make sure that everything is properly connected and nothing is broken by e.g. changes in the classes up in the hierarchy.

Comment (tkornuta-nvidia) [reply Jenkinsfile:327]: added `exp_manager.root dir`, changed to L0 (as it is really integration test)

Comment (tkornuta-nvidia) [reply Jenkinsfile:329]: The script is not using exp_manager and I see no point in adding it - just for test.
So reverting to `rm -rf outputs` - as far as I understand it solves the problem of removing checkpoints.
Will solve the "parallel ci runs" when I will have more tests down there.
